### PR TITLE
Port to oauth2, fix #606

### LINKED ISSUE #606: Authenticate Terraform using GCE metadata tokens
Ensure Terraform has the same behavior as packer in this regard, see part 1 of https://github.com/mitchellh/packer/pull/1679

I will submit a PR porting this change to Terraform

### PR DESCRIPTION
**Pull request by sparkprime**
Terraform equivalent of part 1 of https://github.com/mitchellh/packer/pull/1679

One question:  If the account_file is not given it should attempt to get tokens from the metadata server.  However I cannot figure out how to avoid this prompt:

    provider.google.account_file
      Enter a value: 

At the moment you have to explicitly specify the empty string in the config to avoid the prompt.  In my opinion this is fine but is it what you want?

### CODE REVIEW

**Comment by phinze** [comment]
Alright now that #901 has landed, I _think_ if you rebase on master then all should be well!

**Comment by sparkprime** [comment]
Can one of you comment on my original question about the prompt? thanks

**Comment by phinze** [comment]
> At the moment you have to explicitly specify the empty string in the config to avoid the prompt. In my opinion this is fine but is it what you want?

Seems fine to me too, and you mention it in the docs. :+1:

File: builtin/providers/google/config.go [hunk 1]
```diff
@@ -7,9 +7,12 @@ import (
...
 	"os"
 
-	"code.google.com/p/goauth2/oauth"
-	"code.google.com/p/goauth2/oauth/jwt"
 	"code.google.com/p/google-api-go-client/compute/v1"
+	// oauth2 "github.com/rasa/oauth2-fork-b3f9a68"
+	"github.com/rasa/oauth2-fork-b3f9a68"
+
+	// oauth2 "github.com/rasa/oauth2-fork-b3f9a68/google"
+	"github.com/rasa/oauth2-fork-b3f9a68/google"
```
Comment (phinze) [inline builtin/providers/google/config.go]: Can we verify that this fork is still necessary? I found https://github.com/mitchellh/packer/commit/10dee1bd686e7443f7c4aca90cfcbfad970bf0f9 but I can't seem to find any commentary on what exactly broke and what needs to be fixed upstream.

Comment (sparkprime) [inline builtin/providers/google/config.go]: I am not sure.  I tried github.com/golang/oauth2/google and couldn't work out how to make the build systems pull in the dependency.  When I saw packer using the other one I just copied that and it worked.  I'd like to know the full story as well.

**Comment by sparkprime** [comment]
@phinze re: comment earlier about whether the fork is still necessary, this is upstream (copied from the Packer PR), but it won't build (updatedeps) and I'm not sure why.  Any ideas? thanks

**Comment by sparkprime** [comment]
@lamielle Oops I forgot to tag you.  Better late than never!

**Comment by lamielle** [comment]
Thanks @sparkprime!  Glad to see this PR!

**Comment by sparkprime** [comment]
@phinze Thanks for the pointers, seems to be OK now.  The same modifications could (should?) be made to Packer.

**Comment by phinze** [comment]
Nice! Much cleaner. :shower:

**Pull request merged**